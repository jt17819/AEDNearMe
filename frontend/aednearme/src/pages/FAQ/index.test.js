import { render, fireEvent, screen, cleanup, within } from '@testing-library/react';
import { MemoryRouter, Router as Router } from 'react-router-dom';
import ReactDOM from 'react-dom';
import { createMemoryHistory } from 'history'
import {default as FAQ} from './FAQ';
import axios from 'axios'


describe('FAQ', () => {

  // beforeEach(() => {
  //   render(<App />, { wrapper: MemoryRouter })
  // });


  // Rendering

  it('renders welcome message', () => {
    render(<FAQ />, { wrapper: MemoryRouter })
    expect(screen.getByText('Frequently asked questions')).toBeInTheDocument();
  });

  test('list contains 2 buttons', () => {
    render(<FAQ />, { wrapper: MemoryRouter });
  
    // const listElement = screen.getAllByRole('button');
    const listItems = screen.getAllByRole('heading');
  
    
    expect(listItems.length).toEqual(1);
  });



  // Snapshot

  it('should take a snapshot', () => {
    render(<FAQ />, { wrapper: MemoryRouter })
    const { asFragment } = render(<FAQ />, { wrapper: MemoryRouter })
    
    expect(asFragment(<FAQ />)).toMatchSnapshot()
  })

  

  
})
