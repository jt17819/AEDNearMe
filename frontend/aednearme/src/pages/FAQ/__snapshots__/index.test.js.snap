// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAQ should take a snapshot 1`] = `
<DocumentFragment>
  <h1>
    Frequently asked questions
  </h1>
  <p
    class="MuiTypography-root MuiTypography-body1"
  />
</DocumentFragment>
`;
